app/controllers/resources_controller.rb

```diff
@@ -127,26 +127,43 @@ def file_identifier(object_id, resource_id)
     "#{ID_NAMESPACE}/file/#{object_id.delete_prefix('druid:')}-#{resource_id}/#{SecureRandom.uuid}"
   end
 
-  # rubocop:disable Metrics/AbcSize
-  def decorate_file(file:, version:, external_id: nil)
-    blob = blob_for_signed_id(file.delete(:externalIdentifier), file[:filename])
-    file[:externalIdentifier] = external_id if external_id
-    file[:version] = version
+  def metadata_for_blob(blob, file)
+    file.delete(:externalIdentifier)
     file[:size] = blob.byte_size
     file[:hasMimeType] = blob.content_type || 'application/octet-stream'
     declared_md5 = file[:hasMessageDigests].find { |digest| digest.fetch(:type) == 'md5' }.fetch(:digest)
     calculated_md5 = base64_to_hexdigest(blob.checksum)
     raise BlobError, "MD5 mismatch for #{file[:filename]}" if declared_md5 != calculated_md5
   end
-  # rubocop:enable Metrics/AbcSize
+
+  def metadata_for_file(globus_file, file)
+    return unless File.exist?(globus_file)
+
+    file[:size] = File.size(globus_file)
+  end
+
+  def decorate_file(file:, version:, external_id: nil)
```

Comment:
I'm very late to the party here, but I have no idea what "decorate_file" is doing.  Could a comment be added?

Reply:
Sure.